cmd/clusterctl/client/cluster/mover.go

```diff
@@ -105,6 +113,102 @@ func (o *objectMover) Move(namespace string, toCluster Client, dryRun bool) erro
 	return nil
 }
 
+func (o *objectMover) Save(namespace string, directory string) error {
+	log := logf.Log
+	log.Info("Performing save...")
+
+	objectGraph := newObjectGraph(o.fromProxy)
+
+	// Gets all the types defines by the CRDs installed by clusterctl plus the ConfigMap/Secret core types.
+	err := objectGraph.getDiscoveryTypes()
+	if err != nil {
+		return err
+	}
+
+	// Discovery the object graph for the selected types:
+	// - Nodes are defined the Kubernetes objects (Clusters, Machines etc.) identified during the discovery process.
+	// - Edges are derived by the OwnerReferences between nodes.
+	if err := objectGraph.Discovery(namespace); err != nil {
+		return err
+	}
+
+	// Checks if Cluster API has already completed the provisioning of the infrastructure for the objects involved in the save operation.
+	// This is required because if the infrastructure is provisioned, then we can reasonably assume that the objects we are saving are
+	// not currently waiting for long-running reconciliation loops, and so we can safely rely on the pause field on the Cluster object
+	// for blocking any further object reconciliation on the source objects.
+	if err := o.checkProvisioningCompleted(objectGraph); err != nil {
+		return err
+	}
+
+	// Check whether nodes are not included in GVK considered for move
+	objectGraph.checkVirtualNode()
+
+	if err := o.save(objectGraph, directory); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *objectMover) Restore(namespace string, glob string, directory string) error {
+	log := logf.Log
+	log.Info("Performing restore...")
+
+	objectGraph := newObjectGraph(o.fromProxy)
+
+	// Rebuild the restore objectgraph from files (based on tag / glob / directory)
+	objectsMatcher := regexp.MustCompile(`^.*` + glob + `.*`)
+
+	files, err := filepath.Glob(directory + "/*")
+	if err != nil {
+		return err
+	}
+
+	var rawYamls [][]byte
+	for _, objectFile := range files {
+		if objectsMatcher.MatchString(objectFile) {
+			byObj, err := ioutil.ReadFile(objectFile)
+			if err != nil {
+				return err
+			}
+
+			rawYamls = append(rawYamls, byObj)
+		}
+	}
+
+	processedYamls := yaml.JoinYaml(rawYamls...)
+
+	objs, err := yaml.ToUnstructured(processedYamls)
+	if err != nil {
+		return err
+	}
+
+	objectGraph.getDiscoveryFileTypes(objs)
```

Comment:
Should this be assigned to a variable, or are there just side effects from this call. Might be good to add a comment to call that out if so. Will at least need a `// nolint:errcheck` to make the linter happy.

Reply:
Good catch! Yes, this needs the error check - thanks Sean!!